### PR TITLE
Feat: some pages add  button permission

### PR DESCRIPTION
**Pull request by wangbow**
Signed-off-by: wb-wb665667 <wb-wb665667@alibaba-inc.com>

### Description of your changes

Fixes #Feat: user and roles add button permission

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [x] Run `yarn lint` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: src/pages/Roles/index.tsx [hunk 1]
```diff
@@ -165,29 +166,33 @@ class Roles extends Component<Props, State> {
...
-                size={'medium'}
-                ghost={true}
-                component={'a'}
-                onClick={() => {
-                  this.onDelete(record);
-                }}
-              >
-                <Translation>Delete</Translation>
-              </Button>
+              <Permission request={{ resource: 'role:*', action: 'edit' }} project={''}>
```
Comment (barnettZQG) [inline src/pages/Roles/index.tsx:169]:
```suggestion
              <Permission request={{ resource: `role:${record.name}`, action: 'edit' }} project={''}>
```

File: src/pages/Addons/components/detail/index.tsx [hunk 2]
```diff
@@ -292,31 +293,38 @@ class AddonDetailDialog extends React.Component<Props, State> {
...
-        disabled={status == 'enabling' || status == 'disabling' || version == ''}
-      >
-        <Translation>{status === 'enabled' ? 'Disable' : 'Enable'}</Translation>
-      </Button>,
+      <Fragment>
+        <Button type="secondary" onClick={this.onClose} style={{ marginRight: '16px' }}>
+          <Translation>Cancel</Translation>
+        </Button>
+        ,
+        <Permission request={{ resource: `addon:${addonName}`, action: 'update' }} project={''}>
```
Comment (barnettZQG) [inline src/pages/Addons/components/detail/index.tsx:301]: action: enable or disable

File: src/pages/Addons/index.tsx [hunk 3]
```diff
@@ -101,10 +102,19 @@ class Addons extends React.Component<Props, State> {
         <Title
           title="Addons"
           subTitle="Manages and extends platform capabilities"
-          addButtonTitle={i18n.t('Addon Registries').toString()}
-          addButtonClick={() => {
-            this.setState({ showRegistryManage: true });
-          }}
+          extButtons={[
+            <Permission request={{ resource: 'addon:*', action: 'create' }} project={''}>
```
Comment (barnettZQG) [inline src/pages/Addons/index.tsx:106]: resource: addonRegistry:*

File: src/pages/Cluster/components/CloudServiceDialog/index.tsx [hunk 4]
```diff
@@ -248,16 +249,21 @@ class CloudServiceDialog extends React.Component<Props, State> {
...
           return (
-            <Button
-              text
-              ghost={true}
-              component={'a'}
-              onClick={() => {
-                this.connectcloudCluster(record);
-              }}
+            <Permission
+              request={{ resource: `cluster/namespace:${record.name}`, action: 'create' }}
```
Comment (barnettZQG) [inline src/pages/Cluster/components/CloudServiceDialog/index.tsx:253]: resource: cluster:*

File: src/pages/EnvPage/components/List/index.tsx [hunk 5]
```diff
@@ -121,32 +122,42 @@ class TableList extends Component<Props> {
...
-                      ),
-                      onOk: () => {
-                        this.onDelete(record);
-                      },
-                      locale: locale.Dialog,
-                    });
-                  }}
+                <Permission
+                  request={{ resource: `environment:${record.name}`, action: 'delete' }}
+                  project={'?'}
```
Comment (barnettZQG) [inline src/pages/EnvPage/components/List/index.tsx:127]:
```suggestion
                  project={record.project.name}
```

File: src/pages/EnvPage/components/List/index.tsx [hunk 6]
```diff
@@ -121,32 +122,42 @@ class TableList extends Component<Props> {
...
+                        locale: locale.Dialog,
+                      });
+                    }}
+                  >
+                    <Translation>Remove</Translation>
+                  </a>
+                </Permission>
+                <Permission
+                  request={{ resource: `environment:${record.name}`, action: 'update' }}
+                  project={'?'}
```
Comment (barnettZQG) [inline src/pages/EnvPage/components/List/index.tsx:150]:
```suggestion
                  project={record.project.name}
```

File: src/pages/Users/index.tsx [hunk 7]
```diff
@@ -315,53 +316,73 @@ class Users extends Component<Props, State> {
...
-              <Button
-                text
-                size={'medium'}
-                ghost={true}
-                component={'a'}
-                onClick={() => {
-                  this.onChangeStatus(record);
-                }}
+              <Permission
+                request={{ resource: `user:${record.name}`, action: 'update' }}
```
Comment (barnettZQG) [inline src/pages/Users/index.tsx:354]: action: disable or enable

File: src/pages/Roles/index.tsx (hunk 1 above)
Comment (wangbow) [reply src/pages/Roles/index.tsx]: done

File: src/pages/Addons/components/detail/index.tsx (hunk 2 above)
Comment (wangbow) [reply src/pages/Addons/components/detail/index.tsx]: done

File: src/pages/Addons/index.tsx (hunk 3 above)
Comment (wangbow) [reply src/pages/Addons/index.tsx]: done

File: src/pages/Cluster/components/CloudServiceDialog/index.tsx (hunk 4 above)
Comment (wangbow) [reply src/pages/Cluster/components/CloudServiceDialog/index.tsx]: done

File: src/pages/EnvPage/components/List/index.tsx (hunk 5 above)
Comment (wangbow) [reply src/pages/EnvPage/components/List/index.tsx]: done

File: src/pages/EnvPage/components/List/index.tsx (hunk 6 above)
Comment (wangbow) [reply src/pages/EnvPage/components/List/index.tsx]: done

File: src/pages/Addons/index.tsx [hunk 8]
```diff
@@ -101,10 +102,19 @@ class Addons extends React.Component<Props, State> {
         <Title
           title="Addons"
           subTitle="Manages and extends platform capabilities"
-          addButtonTitle={i18n.t('Addon Registries').toString()}
-          addButtonClick={() => {
-            this.setState({ showRegistryManage: true });
-          }}
+          extButtons={[
+            <Permission request={{ resource: 'addonRegistry:*', action: 'create' }} project={''}>
```
Comment (barnettZQG) [inline src/pages/Addons/index.tsx:106]: action: list

**Review by barnettZQG — APPROVED** [review]
lgtm

**Pull request merged**